test/e2e_new/broker_conformance_test.go

```diff
@@ -72,3 +94,254 @@ func BrokerCreateConfigMap(configName string) *feature.Feature {
+	fs := &feature.FeatureSet{
+		Name: "Knative Kafka Broker Specification - Data Plane",
+		Features: []*feature.Feature{
+			BrokerConsumer(brokerName),
+		},
+	}
+
+	return fs
+}
+
+func BrokerConsumer(brokerName string) *feature.Feature {
+	f := feature.NewFeatureNamed("Ingress")
+
+	sink1 := feature.MakeRandomK8sName("sink-1")
+	sink2 := feature.MakeRandomK8sName("sink-2")
+	sinkTransformer := feature.MakeRandomK8sName("sink-transformer")
+	trigger1 := feature.MakeRandomK8sName("trigger-1")
+	trigger2 := feature.MakeRandomK8sName("trigger-2")
+	triggerTransformer := feature.MakeRandomK8sName("trigger-transformer")
+	triggerReply := feature.MakeRandomK8sName("trigger-reply")
+
+	f.Setup("Set names", func(ctx context.Context, t feature.T) {
+		state.SetOrFail(ctx, t, "brokerName", brokerName)
+		state.SetOrFail(ctx, t, "sink1", sink1)
+		state.SetOrFail(ctx, t, "sink2", sink2)
+	})
+
+	f.Setup("install sink-1", eventshub.Install(sink1, eventshub.StartReceiver))
+	f.Setup("install sink-2", eventshub.Install(sink2, eventshub.StartReceiver))
+	f.Setup("install sink-transformer", eventshub.Install(sinkTransformer,
+		eventshub.ReplyWithTransformedEvent(transformedEventType, transformedEventSource, transformedData),
+		eventshub.StartReceiver),
+	)
+
+	filter1 := eventingv1.TriggerFilterAttributes{
+		"type":   eventingv1.TriggerAnyFilter,
+		"source": eventingv1.TriggerAnyFilter,
+	}
+
+	f.Setup("install trigger1", trigger.Install(
+		trigger1,
+		trigger.WithBrokerName(brokerName),
+		trigger.WithFilter(filter1),
+		trigger.WithSubscriber(service.AsKReference(sink1), ""),
+	))
+	f.Setup("trigger1 goes ready", trigger.IsReady(trigger1))
+
+	filter2 := eventingv1.TriggerFilterAttributes{
+		"source": filteredEventSource,
+	}
+
+	f.Setup("install trigger2", trigger.Install(
+		trigger2,
+		trigger.WithBrokerName(brokerName),
+		trigger.WithFilter(filter2),
+		trigger.WithSubscriber(service.AsKReference(sink2), ""),
+	))
+	f.Setup("trigger2 goes ready", trigger.IsReady(trigger2))
+
+	filterTransformer := eventingv1.TriggerFilterAttributes{
+		"source": replySource,
+		"type":   defaultEventType,
+	}
+
+	f.Setup("install trigger transformer", trigger.Install(
+		triggerTransformer,
+		trigger.WithBrokerName(brokerName),
+		trigger.WithFilter(filterTransformer),
+		trigger.WithSubscriber(service.AsKReference(sinkTransformer), ""),
+	))
+	f.Setup("trigger transformer goes ready", trigger.IsReady(triggerTransformer))
+
+	filterReply := eventingv1.TriggerFilterAttributes{
+		"source": transformedEventSource,
+		"type":   transformedEventType,
+	}
+
+	f.Setup("install trigger reply", trigger.Install(
+		triggerReply,
+		trigger.WithBrokerName(brokerName),
+		trigger.WithFilter(filterReply),
+		trigger.WithSubscriber(service.AsKReference(sink1), ""),
+	))
+	f.Setup("trigger transformer goes ready", trigger.IsReady(triggerTransformer))
+
+	f.Stable("Conformance").
+		ShouldNot("The Broker SHOULD NOT upgrade the CloudEvents version",
```

Comment:
I'm just not sure about using `ShouldNot/Should/Must` in this whole section. Any suggestions?

Reply:
the test should follow the spec language, however, the way we run the tests should and should not in practice are like must / must not as we don't "skip should / should not"